python/tk_desktop/rpc.py

```diff
@@ -131,22 +233,21 @@ def run(self):
                 ready = len(rd) > 0
 
             if not ready:
-                # nothing ready, see if we need to stop the server, if not keep waiting
-                if self._stop:
-                    break
+                logger.debug("multi server thread could not create server")
                 continue
 
-            # connection waiting to be read, accept it
-            connection = self.server.accept()
-            logger.debug("server accepted connection")
             try:
-                while True:
+                # connection waiting to be read, accept it
+                logger.debug("multi server about to accept connection")
+                connection = self.server.accept()
+                logger.debug("multi server accepted connection")
+                while self._is_closed is False:
```

Comment:
When coming at the top of the loop, there's no need to poll if the port is closed. This allows us to use smaller timeouts in the tests.